Update deprecated Sphinx dependencies

GitHub actions for building/deploying Sphinx are currently failing with the following error:

```
Sphinx version error:
The sphinxcontrib.applehelp extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version.
make: 
*** [Makefile:20: html] Error 2
Error: Process completed with exit code 2.
```

The Sphinx versions require updating in the `env.yml` and in the [publish_wps.yml](https://github.com/MetOffice/simulation-systems/blob/main/.github/workflows/publish_wps.yml) GitHub action.

--- a/.github/workflows/publish_wps.yml
+++ b/.github/workflows/publish_wps.yml
@@ -18,14 +18,14 @@ jobs:
         git config --global user.email "umsysteam@metoffice.gov.uk"
         git config --global user.name "SSD Developers"
     - name: Check out source
-      uses: actions/checkout@v3
-    - uses: actions/setup-python@v4
+      uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
       with:
         python-version: '3.x'
     - name: Install Sphinx packages
       run: |
-         pip install Sphinx==4.5.0
-         pip install sphinx-design==0.0.13
+         pip install Sphinx==6.2.1
+         pip install sphinx-design==0.5.0
          pip install sphinx-rtd-theme==1.2.0
          
     - name: build docs

--- a/env.yml
+++ b/env.yml
@@ -1,5 +1,5 @@
 name: sphinx_doc_env
 dependencies:
-  - Sphinx==4.5.0
-  - sphinx-design==0.0.13
+  - Sphinx==6.2.1
+  - sphinx-design==0.5.0
   - sphinx-rtd-theme==1.2.0

--- a/source/conf.py
+++ b/source/conf.py
@@ -18,7 +18,7 @@
 # -- Project information -----------------------------------------------------
 
 project = 'Simulation Systems'
-copyright = '2023, Simulation Systems and Deployment Team'
+copyright = '2023-2024, Simulation Systems and Deployment Team'
 author = 'Simulation Systems and Deployment Team'
 
 

